In [1]:
import numpy as np
from sklearn import linear_model
from sklearn.metrics import mean_squared_error
import math
import pandas as pd

## Reading CSV file 

In [2]:
def Linear_reg(x,y):
    reg = linear_model.LinearRegression()
    reg.fit (x, y)
    print ('Coeffcient:',reg.coef_[0][0])
    print ('Intercept:',reg.intercept_[0])

In [3]:
df=pd.read_csv('/Users/egouda/Documents/DSE/capstone/Combined_merge_EDA.csv')

In [4]:
df_female_child=df[(df['Patient Sex']=='F') & (df['Accurate Age']<16)]

In [5]:
df_male_child = df[(df['Patient Sex']=='M') & (df['Accurate Age']<16)]

In [6]:
df_female_adult=df[(df['Patient Sex']=='F') & (df['Accurate Age']>=16)]

In [7]:
df_male_adult=df[(df['Patient Sex']=='M') & (df['Accurate Age']>=16)]

## Diastole Models

### Female Child less than 16 years

In [8]:
Linear_reg(df_female_child ['Accurate Age'].to_frame(),df_female_child['Diastole'].to_frame())

Coeffcient: 6.99840589281
Intercept: 36.094889496


 \begin{align*}
 f(x) &= 7x+36 \\
\end{align*}

### Male Child less than 16 years


In [9]:
Linear_reg(df_male_child ['Accurate Age'].to_frame(),df_male_child['Diastole'].to_frame())

Coeffcient: 10.5063619276
Intercept: 12.5776705303


 \begin{align*}
 f(x) &= 10.5x+12.5 \\
\end{align*}

### Female Adult

In [10]:
Linear_reg(df_female_adult ['Accurate Age'].to_frame(),df_female_adult['Diastole'].to_frame())


Coeffcient: -0.1414550467
Intercept: 156.483322832


 \begin{align*}
 f(x) &= -0.1x+156.4 \\
\end{align*}

### Male Adult

In [11]:
Linear_reg(df_male_adult ['Accurate Age'].to_frame(),df_male_adult['Diastole'].to_frame())

Coeffcient: -0.32377343594
Intercept: 204.038556426


 \begin{align*}
 f(x) &= -0.3x+204 \\
\end{align*}

## Systole Models

### Female Child less than 16 years

In [12]:
Linear_reg(df_female_child ['Accurate Age'].to_frame(),df_female_child['Systole'].to_frame())

Coeffcient: 1.72868254533
Intercept: 26.0042498438


 \begin{align*}
 f(x) &= 1.7x+26 \\
\end{align*}

### Male Child less than 16 years


In [13]:
Linear_reg(df_male_child ['Accurate Age'].to_frame(),df_male_child['Systole'].to_frame())

Coeffcient: 3.71751563762
Intercept: 6.84799806595


 \begin{align*}
 f(x) &= 3.7x+6.8 \\
\end{align*}

### Female Adult

In [14]:
Linear_reg(df_female_adult ['Accurate Age'].to_frame(),df_female_adult['Systole'].to_frame())

Coeffcient: 0.0968745955659
Intercept: 58.1646107081


 \begin{align*}
 f(x) &= 0.09x+58 \\
\end{align*}

### Male Adult

In [15]:
Linear_reg(df_male_adult ['Accurate Age'].to_frame(),df_male_adult['Systole'].to_frame())

Coeffcient: 0.110315748482
Intercept: 79.1313936401


 \begin{align*}
 f(x) &= 0.11x+79.1\\
\end{align*}